### PR TITLE
applications: nrf_desktop: NVM instead of flash in doc

### PR DESCRIPTION
**Pull request by MarekPieta**
Change switches to using non-volatile memory term instead of flash in the documentation. Other non-volatile memory may be used aswell.

Jira: NCSDK-24793

### CODE REVIEW

**Review by kapi-no — APPROVED** [review]

**Review by mkapala-nordic — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -200,6 +200,8 @@ nRF Desktop
   * The dependencies of the :kconfig:option:`CONFIG_DESKTOP_BLE_LOW_LATENCY_LOCK` Kconfig option.
     The option can be enabled even when the Bluetooth controller is not enabled as part of the application that uses :ref:`nrf_desktop_ble_latency`.
   * The :ref:`nrf_desktop_bootloader` and :ref:`nrf_desktop_bootloader_background_dfu` sections in the nRF Desktop documentation to explicitly mention the supported DFU configurations.
+  * Documentation describing :ref:`nrf_desktop_memory_layout` configuration to simplify gettings started for new users.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:203]:
```suggestion
  * The documentation describing the :ref:`nrf_desktop_memory_layout` configuration to simplify the process of getting started with the application.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -200,6 +200,8 @@ nRF Desktop
...
+  * Changed the flash memory term to the non-volatile memory term to generalize the documentation.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:204]:
```suggestion

* Changed the term flash memory to non-volatile memory to generalize the documentation.
```

File: applications/nrf_desktop/README.rst [hunk 3]
```diff
@@ -1958,8 +1958,8 @@ For this purpose, make sure to enable :kconfig:option:`CONFIG_IMG_MANAGER` and :
 These options allow the :ref:`nrf_desktop_dfu`, :ref:`nrf_desktop_ble_smp`, and :ref:`nrf_desktop_dfu_mcumgr` to manage the DFU image.
 
 .. note::
-  When the MCUboot bootloader is in the swap mode, it can use a secondary image slot located on the external flash.
-  For details on using an external flash for the secondary image slot, see the :ref:`nrf_desktop_pm_external_flash` section.
+  When the MCUboot bootloader is in the swap mode, it can use a secondary image slot located on the external non-volatile memory.
+  For details on using an external non-volatile memory for the secondary image slot, see the :ref:`nrf_desktop_pm_external_flash` section.
```
Comment (peknis) [inline applications/nrf_desktop/README.rst:1962]:
```suggestion
  For details on using external non-volatile memory for the secondary image slot, see the :ref:`nrf_desktop_pm_external_flash` section.
```

File: applications/nrf_desktop/README.rst [hunk 4]
```diff
@@ -1890,7 +1890,7 @@ The nRF Desktop application can use one of the following bootloaders:
 
   * :ref:`Background DFU <nrf_desktop_bootloader_background_dfu>`.
     In this scenario, the MCUboot either swaps the application images located on the secondary and primary slot before booting a new image (``swap mode``) or boots a new application image directly from the secondary image slot (``direct-xip mode``).
-    The swap operation significantly increases boot time after a successful image transfer, but an external flash can be used as the secondary image slot.
+    The swap operation significantly increases boot time after a successful image transfer, but an external non-volatile memory can be used as the secondary image slot.
```
Comment (peknis) [inline applications/nrf_desktop/README.rst:1893]:
```suggestion
    The swap operation significantly increases boot time after a successful image transfer, but external non-volatile memory can be used as the secondary image slot.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 2 above)
Comment (annwoj) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:204]: Would suggest: 
```suggestion
  * Changed the term *flash memory* to *non-volatile memory* for generalization purposes.
```

**Review by mkapala-nordic — APPROVED** [review]

**Comment by MarekPieta** [comment]
Updated reference in `release-notes-1.4.0.rst` to fix doc build failure

**Review by peknis — APPROVED** [review]

**Review by annwoj — APPROVED** [review]

**Review by kapi-no — APPROVED** [review]

**Pull request merged**